src/scripts/main.js

```diff
@@ -1,3 +1,223 @@
 
-// write your code here
+const button = document.querySelector('button');
+const messageStart = document.querySelector('.message-start');
+const messageWin = document.querySelector('.message-win');
+const messageLose = document.querySelector('.message-lose');
+const gameScore = document.querySelector('.game-score');
+const fieldRows = document.querySelectorAll('.field-row');
+const size = 4;
+let score = 0;
+let isWin = false;
+let newGameField;
+let gameField = [
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+  [0, 0, 0, 0],
+];
+
+button.addEventListener('click', e => {
+  document.addEventListener('keydown', move);
+
+  if (button.classList.contains('start')) {
+    button.classList.replace('start', 'restart');
+    button.innerText = 'Restart';
+    messageStart.classList.add('hidden');
+  } else {
+    isWin = false;
+    reset();
+  }
+
+  addNumber();
+  addNumber();
+  render();
+});
+
+function move(e) {
+  newGameField = gameField;
+
+  switch (e.key) {
+    case 'ArrowLeft':
+      left();
+      break;
+
+    case 'ArrowRight':
+      right();
+      break;
+
+    case 'ArrowDown':
+      down();
+      break;
+
+    case 'ArrowUp':
+      up();
+      break;
+
+    default:
+      return;
+  }
+
+  for (let row = 0; row < size; row++) {
+    for (let col = 0; col < size; col++) {
+      if (newGameField[row][col] !== gameField[row][col]) {
+        gameField = newGameField;
+        addNumber();
+        render();
+      }
+    }
+  }
+
+  if (isWin) {
+    messageWin.classList.remove('hidden');
+  }
+
+  if (!isPosibleToMove()) {
+    messageLose.classList.remove('hidden');
+    document.removeEventListener('keydown', move);
+  }
+}
+
+function reset() {
+  score = 0;
+
+  gameField = [
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+    [0, 0, 0, 0],
+  ];
+
+  if (!messageLose.classList.contains('hidden')) {
+    messageLose.classList.add('hidden');
+  }
+
+  if (!messageWin.classList.contains('hidden')) {
+    messageWin.classList.add('hidden');
+  }
+};
+
+function addNumber() {
+  const [randomY, randomX] = findEmptyCell();
+
+  gameField[randomY][randomX] = Math.random() < 0.9 ? 2 : 4;
+};
+
+function findEmptyCell() {
+  const emptyCells = [];
+
+  gameField.forEach((row, rowIndex) => {
+    row.forEach((cell, columnIndex) => {
+      if (cell === 0) {
+        emptyCells.push([rowIndex, columnIndex]);
+      }
+    });
+  });
+
+  return emptyCells[Math.floor(Math.random() * emptyCells.length)];
+}
+
+function render() {
+  for (let row = 0; row < size; row++) {
+    for (let col = 0; col < size; col++) {
+      const elem = fieldRows[row].children[col];
+      const cell = gameField[row][col];
+
+      if (cell === 0) {
+        elem.textContent = '';
+        elem.className = 'field-cell';
+      } else {
+        elem.textContent = cell;
+        elem.className = `field-cell field-cell--${cell}`;
+      }
+    };
+  };
+
+  gameScore.textContent = score;
+};
+
+function left() {
+  if (!checkRows()) {
```

Comment:
The function 'left' is not clearly defined. It is better to use more descriptive function names.